File: backend_js/my_agent/LLMManager.ts
import { ChatOpenAI } from "@langchain/openai";
import { ChatPromptTemplate } from "@langchain/core/prompts";

class LLMManager {
  private llm: ChatOpenAI;

  constructor() {
    this.llm = new ChatOpenAI({
      modelName: "gpt-4o",
      temperature: 0,
    });
    this.invoke = this.invoke.bind(this);
  }

  async invoke(
    prompt: ChatPromptTemplate,
    kwargs: Record<string, any>
  ): Promise<string> {
    const messages = await prompt.formatMessages(kwargs);
    const response = await this.llm.invoke(messages);
    return response.content as string;
  }
}

export default LLMManager;

File: backend_py/my_agent/LLMManager.py
from langchain_core.prompts import ChatPromptTemplate
from langchain_openai import ChatOpenAI

class LLMManager:
    def __init__(self):
        self.llm = ChatOpenAI(model="gpt-4o", temperature=0)

    def invoke(self, prompt: ChatPromptTemplate, **kwargs) -> str:
        messages = prompt.format_messages(**kwargs)
        response = self.llm.invoke(messages)
        return response.content